### PR TITLE
ring: added night mode support to arm with sensor bypass

### PR DESCRIPTION
**Pull request by leedsalex**
closes https://github.com/koush/scrypted/issues/482

This PR also fixes two unrelated warnings I noticed in the homekit logs for the garage and window covering type. The following was not needed as we already call `bindCharacteristic`
```
service.getCharacteristic(Characteristic.TargetDoorState)
.on(CharacteristicEventTypes.GET, (callback: NodeCallback<CharacteristicValue>) => {
    callback(null, targetState);
})
```

### CODE REVIEW

File: plugins/homekit/src/types/garage.ts [hunk 1]
```diff
@@ -22,9 +22,6 @@ addSupportedType({
 
         let targetState = !!device.entryOpen ? Characteristic.TargetDoorState.OPEN : Characteristic.TargetDoorState.CLOSED;
         service.getCharacteristic(Characteristic.TargetDoorState)
-            .on(CharacteristicEventTypes.GET, (callback: NodeCallback<CharacteristicValue>) => {
```
Comment (koush) [inline plugins/homekit/src/types/garage.ts:25]: is this deletion intentional? not sure why garage is being changed.

File: plugins/homekit/src/types/windowcovering.ts [hunk 2]
```diff
@@ -28,9 +28,6 @@ addSupportedType({
         let targetState = !!device.entryOpen ? 100 : 0;
         service.getCharacteristic(Characteristic.TargetPosition)
             .setProps(props)
-            .on(CharacteristicEventTypes.GET, (callback: NodeCallback<CharacteristicValue>) => {
```
Comment (koush) [inline plugins/homekit/src/types/windowcovering.ts:31]: what's this?

**Comment by koush** [comment]
ring changes look fine, but I dont know what some of those homekit changes are (one looks like a brevity cleanup)

File: plugins/homekit/src/types/garage.ts (hunk 1 above)
Comment (leedsalex) [reply plugins/homekit/src/types/garage.ts:25]: I made a comment in the pr description. Those are unneeded, as we are calling the callback twice per event. Once from bind characteristic and then again here. But the bind characteristic handling is much better. The double callback is causing warning all over the logs.

File: plugins/homekit/src/types/windowcovering.ts (hunk 2 above)
Comment (leedsalex) [reply plugins/homekit/src/types/windowcovering.ts:31]: Same as above

**Pull request merged**